### PR TITLE
refactor: remove parking-lot dependency

### PR DESCRIPTION
**Pull request by dignifiedquire**
## Description

Currently there was an inconsistent usage of `std::sync::Mutex` and `parking_lot::Mutex`. This normalizes the usage to always use `std::sync::Mutex` and remove the external dependency.

## Change checklist

- [x] Self-review.
- [x] Documentation updates following the [style guide](https://rust-lang.github.io/rfcs/1574-more-api-documentation-conventions.html#appendix-a-full-conventions-text), if relevant.
- [x] Tests if relevant.
- [x] All breaking changes documented.

### CODE REVIEW

File: iroh/src/discovery.rs [hunk 1]
```diff
@@ -500,6 +499,7 @@ mod tests {
             self.shared
                 .nodes
                 .lock()
+                .unwrap()
```
Comment (flub) [inline iroh/src/discovery.rs:502]: I would prefer if we used `.expect("poisoned")` as that way we can more easily treat `.unwrap()` as "un-audited".  (Having done this whole thing about auditing all our unwraps once before :wink: )

**Review by flub — COMMENTED** [review]
I'm all for this but would prefer `.expect("poisoned")` over `.unwrap()`.

**Comment by dignifiedquire** [comment]
> I'm all for this but would prefer `.expect("poisoned")` over `.unwrap()`.

we have never done this, and that a lot of unwraps like this that don't have that.. we can do it in one go as a follow up if you really want to

**Comment by dignifiedquire** [comment]
seems we don't have many `unwrap`s.., okay I will change them

File: iroh/src/discovery.rs (hunk 1 above)
Comment (dignifiedquire) [reply iroh/src/discovery.rs:501]: fixed

**Comment by flub** [comment]
> > I'm all for this but would prefer `.expect("poisoned")` over `.unwrap()`.
> 
> we have never done this, and that a lot of unwraps like this that don't have that.. we can do it in one go as a follow up if you really want to

For completeness, we have done this:
https://github.com/n0-computer/iroh/issues/1528
https://github.com/n0-computer/iroh/pull/2046

**Comment by dignifiedquire** [comment]
yep, we just missed adding the lints, which is why they can creep back in

File: iroh/src/magicsock/node_map.rs [hunk 2]
```diff
@@ -762,7 +799,7 @@ mod tests {
             // add address
             node_map.add_test_addr(node_addr);
             // make it active
-            node_map.inner.lock().receive_udp(addr);
+            node_map.inner.lock().unwrap().receive_udp(addr);
```
Comment (flub) [inline iroh/src/magicsock/node_map.rs:802]: still unwrap

File: iroh/src/magicsock/node_map.rs [hunk 3]
```diff
@@ -772,7 +809,7 @@ mod tests {
             node_map.add_test_addr(node_addr);
         }
 
-        let mut node_map_inner = node_map.inner.lock();
+        let mut node_map_inner = node_map.inner.lock().unwrap();
```
Comment (flub) [inline iroh/src/magicsock/node_map.rs:812]: still unwrap

File: iroh/src/magicsock/node_map.rs [hunk 4]
```diff
@@ -824,6 +866,7 @@ mod tests {
         node_map
             .inner
             .lock()
+            .unwrap()
```
Comment (flub) [inline iroh/src/magicsock/node_map.rs:869]: still unwrap

File: iroh/src/discovery.rs [hunk 5]
```diff
@@ -508,7 +508,7 @@ mod tests {
             node_id: NodeId,
         ) -> Option<BoxStream<Result<DiscoveryItem>>> {
             let addr_info = match self.resolve_wrong {
-                false => self.shared.nodes.lock().get(&node_id).cloned(),
+                false => self.shared.nodes.lock().unwrap().get(&node_id).cloned(),
```
Comment (flub) [inline iroh/src/discovery.rs:511]: still unwrap

File: iroh/src/discovery.rs (hunk 1 above)
Comment (flub) [reply iroh/src/discovery.rs:501]: except this one?

File: iroh/src/magicsock/node_map.rs (hunk 2 above)
Comment (dignifiedquire) [reply iroh/src/magicsock/node_map.rs:802]: that's a test, that's fine

**Review by flub — APPROVED** [review]

**Pull request merged**